secretmanager/src/main/java/com/example/ListSecretVersions.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example;
+
+// [START secretmanager_list_secret_versions]
+import com.google.cloud.secretmanager.v1beta1.ListSecretVersionsRequest;
+import com.google.cloud.secretmanager.v1beta1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1beta1.SecretManagerServiceClient.ListSecretVersionsPagedResponse;
+import com.google.cloud.secretmanager.v1beta1.SecretName;
+import com.google.cloud.secretmanager.v1beta1.SecretVersion;
+import java.io.IOException;
+
+public class ListSecretVersions {
+
+  // List all secret versions for a secret.
+  public void listSecretVersions(String projectId, String secretId) throws IOException {
+    // Create a Secret Manager client with cleanup.
+    try (SecretManagerServiceClient client = SecretManagerServiceClient.create()) {
+      // Build the parent name.
+      SecretName parent = SecretName.of(projectId, secretId);
+
+      // Create the request.
+      ListSecretVersionsRequest request =
+          ListSecretVersionsRequest.newBuilder().setParent(parent.toString()).build();
+
+      // Get all versions, paginated.
+      ListSecretVersionsPagedResponse iterator = client.listSecretVersions(request);
+
+      // List all versions and their state.
```

Comment:
Does this iterate all versions, or just the versions in this page?

Reply:
Umm - I was under the impression that it would iterate over all objects in all pages. Am I misunderstanding? I changed it to use forEach, does that change the behavior?

Reply:
Verified it's all pages. Updated.